src/measures/fermions/ellesettete.cpp

```diff
@@ -113,31 +129,59 @@ namespace nissa
 		    apply_stag_op(g5_id_source,conf,theory_pars.backfield[iflav],GAMMA_INT::GAMMA_5,GAMMA_INT::IDENTITY,source);
 		    apply_stag_op(id_g5_source,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,source);
 
-		    //compute std 2pts propagator G(m|n) ~ [D^-1(m|y) source(y)] source(n)* and simple sequential propagator
-		    MINV(SIMPLE_PROP,iflav,source);
-		    MINV(SEQ_PROP,iflav,SIMPLE_PROP);
-
-		    //compute  2pts propagator with id x g5 at source and apply id x g5 at sink
-		    MINV(PROP_ID_G5,iflav,id_g5_source);
-		    apply_stag_op(ID_G5_PROP_ID_G5,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,PROP_ID_G5);
-
-		    //compute sequential propagator with id x g5 at source and apply id x g5 at sink
-		    MINV(SEQ_PROP_ID_G5,iflav,PROP_ID_G5);
-		    apply_stag_op(ID_G5_SEQ_PROP_ID_G5,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,SEQ_PROP_ID_G5);
-			
-
-		    //then glb reduction to compute the trace for the connected 2pts_iso, 2pts, 3pts and 4pts diagrams
-		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_two_pts_iso,SIMPLE_PROP,SIMPLE_PROP);
-		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_two_pts,SIMPLE_PROP,ID_G5_PROP_ID_G5);
-		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_three_pts,SIMPLE_PROP,ID_G5_SEQ_PROP_ID_G5);
-		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_four_pts,SEQ_PROP,ID_G5_SEQ_PROP_ID_G5);
-
-		    //////// disconnected ////////
-		    //here we need just simple seq prop with nothing at source
-		    if(ihit==meas_pars.nhits-1) master_fprintf(file," # Tr_no_insertion_bubble source time %d\n", glb_t);
-		    SUMM_THE_TRACE_PRINT_AT_LAST_HIT(Tr_no_insertion_bubble,g5_id_source,SIMPLE_PROP);
-		    if(ihit==meas_pars.nhits-1) master_fprintf(file,"\n # Tr_insertion_bubble source time %d\n", glb_t);
-		    SUMM_THE_TRACE_PRINT_AT_LAST_HIT(Tr_insertion_bubble,g5_id_source,SEQ_PROP);
+			if(method_flag)
```

Comment:
`if(method==1)`
but, why don't you put the if at the very beginning, so to avoid defining the traces etc ? in the current position, you will print the analytical traces even if method is numerical, don't you?

Reply:
no it won’t print the analytical traces, and i cannot move them to the analytical part as they would be initialized at each hit step if $nhits \neq 1$. Moreover if i moved the flag checking at the very beginning i would have to duplicate the previous loops (over nhits and glb_t)